### PR TITLE
Clean up calls to FillBoundary in FDTD part of `OneStep_nosub`

### PR DESCRIPTION
**Pull request by EZoni**
Small cleanup of calls to `FillBoundary` for the various fields E, B, F, and G, in the FDTD part of the `OneStep_nosub` function. Details of the few code changes summarized in the comment below: https://github.com/ECP-WarpX/WarpX/pull/3607#discussion_r1065055555.

### CODE REVIEW

File: Source/Evolve/WarpXEvolve.cpp [hunk 1]
```diff
@@ -466,20 +466,22 @@ WarpX::OneStep_nosub (Real cur_time)
...
-
         FillBoundaryE(guard_cells.ng_FieldSolver, WarpX::sync_nodal_points);
+
         EvolveF(0.5_rt * dt[0], DtType::SecondHalf);
         EvolveG(0.5_rt * dt[0], DtType::SecondHalf);
         EvolveB(0.5_rt * dt[0], DtType::SecondHalf); // We now have B^{n+1}
 
         if (do_pml) {
-            FillBoundaryF(guard_cells.ng_alloc_F);
+            //FillBoundaryF(guard_cells.ng_alloc_F);
```
Comment (EZoni) [inline Source/Evolve/WarpXEvolve.cpp]:
```suggestion
```

File: Source/Evolve/WarpXEvolve.cpp [hunk 2]
```diff
@@ -453,8 +453,8 @@ WarpX::OneStep_nosub (Real cur_time)
         EvolveG(0.5_rt * dt[0], DtType::FirstHalf);
         FillBoundaryF(guard_cells.ng_FieldSolverF);
         FillBoundaryG(guard_cells.ng_FieldSolverG);
-        EvolveB(0.5_rt * dt[0], DtType::FirstHalf); // We now have B^{n+1/2}
 
+        EvolveB(0.5_rt * dt[0], DtType::FirstHalf); // We now have B^{n+1/2}
         FillBoundaryB(guard_cells.ng_FieldSolver, WarpX::sync_nodal_points);
```
Comment (EZoni) [inline Source/Evolve/WarpXEvolve.cpp:458]: Here I just grouped the calls to `EvolveB` and `FillBoundaryB` together, no code changes.

File: Source/Evolve/WarpXEvolve.cpp [hunk 3]
```diff
@@ -466,20 +466,21 @@ WarpX::OneStep_nosub (Real cur_time)
...
 
         if (do_pml) {
-            FillBoundaryF(guard_cells.ng_alloc_F);
             DampPML();
             NodalSyncPML();
             FillBoundaryE(guard_cells.ng_MovingWindow);
-            FillBoundaryF(guard_cells.ng_MovingWindow);
             FillBoundaryB(guard_cells.ng_MovingWindow);
+            FillBoundaryF(guard_cells.ng_MovingWindow);
+            FillBoundaryG(guard_cells.ng_MovingWindow);
```
Comment (EZoni) [inline Source/Evolve/WarpXEvolve.cpp:481]: Real code changes:
- removed call to `FillBoundaryF` before `DampPML`/`NodalSyncPML` (already called after those);
- added call to `FillBoundaryG` after `DampPML`/`NodalSyncPML`, like the ones for E, B, and F.

Note that the missing call for G could have been potentially a bug.

Comment (lucafedeli88) [reply Source/Evolve/WarpXEvolve.cpp:481]: Interesting... with which set of parameters do you think that this could have affected WarpX ?

Comment (EZoni) [reply Source/Evolve/WarpXEvolve.cpp:481]: The missing call to `FillBoundaryG` could affect simulations wher div(B) cleaning is used in the regular domain and in the PML (G being the scalar field added to Maxwell's equations when div(B) cleaning is used, such that dG/dt = div(B)). On the other side, I think the first call to `FillBoundaryF` before the PML damping and synchronization, now removed, was simply redundant.

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**